lib/solr/configuration.rb

```diff
@@ -1,16 +1,34 @@
+require 'solr/field'
+require 'solr/dynamic_field_type'
+require 'solr/field_definition_builder'
+require 'solr/type_definition_builder'
+
 module Solr
   class Configuration
-    attr_accessor :field_map, :open_timeout, :read_timeout, :url
+    attr_accessor :open_timeout, :read_timeout, :url, :field_types, :fields
 
     def initialize
-      @field_map = {}
       @read_timeout = 2
       @open_timeout = 8
-      @url = ENV.fetch('SOLR_URL')
+      @url = ENV['SOLR_URL']
+      @field_types = {}
+      @fields = {}
     end
 
     def uri
       @uri ||= URI.parse(url)
     end
+
+    def define_field_types
+      builder = Solr::TypeDefinitionBuilder.new
```

Comment:
please update the builder names correspondingly (to configuration block method changes)